### PR TITLE
Add chart tests

### PR DESCRIPTION
**Pull request by andresmgot**
These simple tests ensures that:

 - The the Nginx proxy and the `dashboard` app are running and return a react error message (due to the lack of javascript).
 - The `apprepository-controller`, `chart-repo` and `chart-svc` work together checking that the chart `wordpress` is present.
 - Tiller-proxy is working and it list the `kubeapps` chart installed.

Execution:

```
▶ time helm test right-woodpecker

RUNNING: right-woodpecker-tiller-proxy-test
PASSED: right-woodpecker-tiller-proxy-test
RUNNING: right-woodpecker-chartsvc-test
PASSED: right-woodpecker-chartsvc-test
RUNNING: right-woodpecker-dashboard-test
PASSED: right-woodpecker-dashboard-test

helm 0.17s user 0.07s system 1% cpu 14.172 total
```

### CODE REVIEW

File: chart/kubeapps/templates/tests/test-chartsvc.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  name: "{{ .Release.Name }}-chartsvc-test"
+  annotations:
+    "helm.sh/hook": test-success
+spec:
+  containers:
+  - name: {{ .Release.Name }}-chartsvc-test
+    image: appropriate/curl
```
Comment (prydonius) [inline chart/kubeapps/templates/tests/test-chartsvc.yaml]: Can we use an image that we have more control over here, such as bitnami/minideb? It's a little bigger, but hopefully not that big of a deal for running `helm test`.

**Comment by andresmgot** [comment]
@prydonius should I land this or there is something I should wait for?

**Comment by prydonius** [comment]
@andresmgot how is CircleCI enabled in this PR?

File: chart/kubeapps/templates/tests/test-chartsvc.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  name: "{{ .Release.Name }}-chartsvc-test"
+  annotations:
+    "helm.sh/hook": test-success
+spec:
+  containers:
+  - name: {{ .Release.Name }}-chartsvc-test
+    image: {{ .Values.frontend.image.repository }}:{{ .Values.frontend.image.tag }}
```
Comment (prydonius) [inline chart/kubeapps/templates/tests/test-chartsvc.yaml]: should we add a chartTests.image value?

File: chart/kubeapps/templates/tests/test-chartsvc.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    env:
+      - name: CHARTSVC_HOST
```
Comment (prydonius) [inline chart/kubeapps/templates/tests/test-chartsvc.yaml]: indentation is mismatched here compared to the containers array, we should have `-` aligned with `name:` above

File: chart/kubeapps/templates/tests/test-chartsvc.yaml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+        value: {{ template "kubeapps.chartsvc.fullname" . }}.{{ .Release.Namespace }}
+      - name: CHARTSVC_PORT
+        value: "{{ .Values.chartsvc.service.port }}"
+    command: ["sh", "-c", "curl $CHARTSVC_HOST:$CHARTSVC_PORT/v1/charts | grep wordpress"]
```
Comment (prydonius) [inline chart/kubeapps/templates/tests/test-chartsvc.yaml]: can we use the YAML array notation, it makes this easier to read

File: chart/kubeapps/templates/tests/test-dashboard.yaml [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+metadata:
+  name: "{{ .Release.Name }}-dashboard-test"
+  annotations:
+    "helm.sh/hook": test-success
+spec:
+  containers:
+  - name: {{ .Release.Name }}-dashboard-test
+    image: {{ .Values.frontend.image.repository }}:{{ .Values.frontend.image.tag }}
+    env:
+      - name: DASHBOARD_HOST
```
Comment (prydonius) [inline chart/kubeapps/templates/tests/test-dashboard.yaml]: indentation

File: chart/kubeapps/templates/tests/test-dashboard.yaml [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+        value: {{ template "kubeapps.fullname" . }}.{{ .Release.Namespace }}
+    command: ["sh", "-c", "curl $DASHBOARD_HOST | grep 'You need to enable JavaScript to run this app.'"]
```
Comment (prydonius) [inline chart/kubeapps/templates/tests/test-dashboard.yaml]: YAML array format

File: chart/kubeapps/templates/tests/test-tiller-proxy.yaml [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+  - name: {{ .Release.Name }}-tiller-proxy-test
+    image: {{ .Values.frontend.image.repository }}:{{ .Values.frontend.image.tag }}
+    env:
+      - name: TILLER_PROXY_HOST
+        value: {{ template "kubeapps.tiller-proxy.fullname" . }}.{{ .Release.Namespace }}
+      - name: TILLER_PROXY_PORT
+        value: "{{ .Values.tillerProxy.service.port }}"
+      - name: KUBEAPPS_RELEASE
+        value: {{ .Release.Name }}
+    command: ["sh", "-c", "curl -ik -H \"Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)\" $TILLER_PROXY_HOST:$TILLER_PROXY_PORT/v1/releases | grep $KUBEAPPS_RELEASE"]
```
Comment (prydonius) [inline chart/kubeapps/templates/tests/test-tiller-proxy.yaml]: YAML array format

File: chart/kubeapps/templates/tests/test-tiller-proxy.yaml [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+metadata:
+  name: "{{ .Release.Name }}-tiller-proxy-test"
+  annotations:
+    "helm.sh/hook": test-success
+spec:
+  containers:
+  - name: {{ .Release.Name }}-tiller-proxy-test
+    image: {{ .Values.frontend.image.repository }}:{{ .Values.frontend.image.tag }}
+    env:
+      - name: TILLER_PROXY_HOST
```
Comment (prydonius) [inline chart/kubeapps/templates/tests/test-tiller-proxy.yaml]: indentation

**Comment by andresmgot** [comment]
> @andresmgot how is CircleCI enabled in this PR?

It was enabled for the whole repo but it doesn't have the `config.yaml` yet so it doesn't work. I disabled it so it won't appear again.

File: chart/kubeapps/templates/tests/test-chartsvc.yaml (hunk 2 above)
Comment (andresmgot) [reply chart/kubeapps/templates/tests/test-chartsvc.yaml]: I don't mind but wouldn't that add noise to the values.yaml? It would be a corner case if someone needs that flexibility

Comment (prydonius) [reply chart/kubeapps/templates/tests/test-chartsvc.yaml]: My only concern is that using frontend's image could be dangerous, it would have a side-effect we might forget about if we change that image. I don't think it's too bad to put it in values.yaml, users may want to pull from a different registry.

**Comment by andresmgot** [comment]
@prydonius pal
